Library/Homebrew/version.rb

```diff
@@ -379,3 +392,56 @@ def self._parse(spec)
     return m.captures.first unless m.nil?
   end
 end
+
+class HeadVersion < Version
+  attr_reader :version
+  attr_reader :commit
+
+  def initialize(val)
+    if val.respond_to?(:to_str)
+      @version = if (val.to_str.start_with?("HEAD"))
```

Comment:
Briefly scanning the codes, I find this condition seems to be true for any cases.